src/network/networkd-sysctl.c

```diff
@@ -40,10 +41,71 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 }
 
+static int sysctl_inotify_callback(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        const char *desc = NULL, *value;
+        Manager *m = userdata;
+        char *path, *buf;
+        size_t len;
+
+        if (sd_event_source_get_description(source, &desc) < 0)
+                return 0;
+
+        path = strjoina(desc, "/", event->name);
+
+        value = hashmap_get(m->sysctls, path);
+
+        /* We don't handle this sysctl */
+        if (!value)
+                return 0;
+
+        if (read_full_file(path, &buf, &len) < 0)
+                return 0;
+
+        if (len && buf[len - 1] == '\n')
+                buf[len - 1] = 0;
+
+        if (strcmp(value, buf)) {
+                path += sizeof("/proc/sys/") - 1;
+                for (char *p = path; *p; p++)
+                        if (*p == '/')
+                                *p = '.';
```

Comment:
use sysctl_normalize()

Reply:
Seems to me that `sysctl_normalize` converts `a.b.c` into `a/b/c`, I'm doing the opposite

Reply:
So instead of `net.ipv4.conf.enp0s1.promote_secondaries` I should log `/proc/sys/net/ipv4/conf/enp0s1/promote_secondaries`?